### PR TITLE
feat: Qdrant as a supported vector store

### PR DESCRIPTION
**Pull request by Anush008**
## Description

This PR adds support for Qdrant - https://qdrant.tech/ to be used as a vector store.

### CODE REVIEW

**Comment by zainhoda** [comment]
Awesome! @Anush008 once we add this, would you be interested in doing a joint blog post?

**Comment by Anush008** [comment]
Hi Zain. We can definitely come up with content to promote this.

File: src/vanna/qdrant/qdrant.py [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+                    id=id,
+                    vector=self.generate_embedding(documentation),
+                    payload={
+                        "documentation": documentation,
+                    },
+                )
+            ],
+        )
+
+        return self._format_point_id(id, DOCUMENTATION_COLLECTION_NAME)
```
Comment (Anush008) [inline src/vanna/qdrant/qdrant.py:133]: Qdrant only allows valid UUIDs and unsigned integers as point IDs.
So, we can't have these suffixes in the IDs. Therefore, we're only appending them when returning the ID.
Which is later trimmed when the ID is used for deleting.

Comment (zainhoda) [reply src/vanna/qdrant/qdrant.py:133]: Understood!

**Comment by zainhoda** [comment]
@Anush008 I added tests and in the process of adding tests I found a couple of places where I had to make some minor changes. Could you review the changes that I made?

File: src/vanna/qdrant/qdrant.py [hunk 2]
```diff
@@ -305,7 +305,7 @@ def _setup_collections(self):
             )
         if not self._client.collection_exists(DOCUMENTATION_COLLECTION_NAME):
             self._client.create_collection(
-                name=DOCUMENTATION_COLLECTION_NAME,
+                collection_name=DOCUMENTATION_COLLECTION_NAME,
```
Comment (Anush008) [inline src/vanna/qdrant/qdrant.py:308]: I had passed this as a positional arg instead. Unsure where the `name` key came from. Thanks.

**Review by zainhoda — APPROVED** [review]

**Pull request merged**